src/javascripts/Views/referencedByView.js

```diff
@@ -0,0 +1,104 @@
+ngapp.controller('referencedByViewController', function($scope, $element, $timeout, htmlHelpers, treeService, referencedByViewService, referencedByViewColumnsService, hotkeyService, nodeSelectionService, treeViewService, treeColumnService, contextMenuService, contextMenuFactory) {
+    // link view to scope
+    $scope.view = $scope.$parent.tab;
+    $scope.view.scope = $scope;
+
+    $scope.allColumns = referencedByViewColumnsService.columns;
+    $scope.contextMenuItems = contextMenuFactory.referencedByViewItems;
+
+    // inherited functions
+    treeService.buildFunctions($scope, $element);
+    treeViewService.buildFunctions($scope, $element);
+    referencedByViewService.buildFunctions($scope);
+    treeColumnService.buildFunctions($scope, '.referenced-by-view', true);
+    nodeSelectionService.buildFunctions($scope, false);
+    hotkeyService.buildOnKeyDown($scope, 'onTreeKeyDown', 'referencedByView');
+
+    $scope.open = function(node) {
+        const record = node.handle ? node.handle : node;
+        $scope.record = record;
+        $scope.syncWithLinkedViews(record);
+    };
+
+    // scope functions
+    $scope.showContextMenu = function(e) {
+        contextMenuService.showContextMenu($scope, e);
+    };
+
+    $scope.toggleAddressBar = function(visible) {
+        $scope.showAddressBar = visible;
+        if (visible) {
+            $timeout($scope.focusAddressInput, 50);
+        } else if ($scope.treeElement) {
+            $scope.treeElement.focus();
+        }
+    };
+
+    $scope.focusAddressInput = function () {
+        let addressInput = htmlHelpers.resolveElement($scope.tabView, 'record-address-bar/input');
+        if (addressInput) addressInput.focus();
+    };
+
+    $scope.openColumnsModal = function() {
+        $scope.$emit('openModal', 'editColumns', {
+            allColumns: $scope.allColumns,
+            columnsService: referencedByViewColumnsService
+        });
+    };
+
+    $scope.onScroll = function(e) {
+        $scope.columnsElement.scrollLeft = e.currentTarget.scrollLeft;
+    };
+
+    $scope.onNodeDoubleClick = function(e, node) {
+        $scope.open(node);
+    };
+
+    $scope.handleEnter = function(e) {
+        $scope.open($scope.lastSelectedNode().handle);
+        e.stopImmediatePropagation();
+    };
+
+    // initialization
+    $scope.$on('setRecord', function(e, record) {
+        $scope.open(record);
+        e.stopPropagation();
+    });
+    $scope.$on('reloadGUI', function() {
+        if (!$scope.record) return;
```

Comment:
if we link this to the record view we should be able to remove all of this logic, as the record view already does it.  The record view will simply set the referenced by view's `$scope.record` from its `$scope.record`.

Reply:
actually, we'll need to keep some of it if we want to allow drag and drop from the tree view.  :thinking:

Reply:
:+1: